File: TextAnalysisFront/src/app/models/AnalysedText.ts
export class AnalysedText {
    private _repeated: string[];
    private _archaisms: string[];
    private _slangs: string[];
    private _irregulars: string[];
    private _expressions: string[];

    public constructor(
        repeated?: string[],
        archaisms?: string[],
        slangs?: string[],
        irregulars?: string[],
        expressions?: string[]
    ) {
        this.repeated = repeated;
        this.archaisms = archaisms;
        this.slangs = slangs;
        this.irregulars = irregulars;
        this.expressions = expressions;
     }

    get repeated():string[]{
        console.debug("get repeated: " + JSON.stringify(this._repeated));
        return this._repeated;
    }

    set repeated(val){
        this._repeated=val;
        console.debug("set repeated: " + JSON.stringify(this._repeated));
    }

    get archaisms():string[]{
        console.debug("get archaisms: " + JSON.stringify(this._archaisms));
        return this._archaisms;
    }

    set archaisms(val){
        this._archaisms=val;
        console.debug("set archaisms: " + JSON.stringify(this._archaisms));
    }

    get slangs():string[]{
        console.debug("get slangs: " + JSON.stringify(this._slangs));
        return this._slangs;
    }

    set slangs(val){
        this._slangs=val;
        console.debug("set slangs: " + JSON.stringify(this._slangs));
    }

    get irregulars():string[]{
        console.debug("get irregulars: " + JSON.stringify(this._irregulars));
        return this._irregulars;
    }

    set irregulars(val){
        this._irregulars=val;
        console.debug("set irregulars: " + JSON.stringify(this._irregulars));
    }

    get expressions():string[]{
        console.debug("get expressions: " + JSON.stringify(this._expressions));
        return this._expressions;
    }

    set expressions(val){
        this._expressions=val;
        console.debug("set expressions: " + JSON.stringify(this._expressions));
    }
}

File: TextAnalysisAndroid/app/src/main/java/com/likhtman/TextAnalysis/Model/AnalysedText.java
package com.likhtman.TextAnalysis.Model;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.HashSet;

public class AnalysedText {
    private HashSet<String> repeated = new HashSet<String>();
    private HashSet<String> archaisms = new HashSet<String>();
    private HashSet<String> slangs = new HashSet<String>();
    private HashSet<String> irregulars = new HashSet<String>();
    private HashSet<String> expressions = new HashSet<String>();

    public AnalysedText() { }

    public AnalysedText(
            HashSet<String> tmpRepeated,
            HashSet<String> tmpArchaisms,
            HashSet<String> tmpSlangs,
            HashSet<String> tmpIrregulars,
            HashSet<String> tmpExpressions)
    {
        repeated.addAll(tmpRepeated);
        archaisms.addAll(tmpArchaisms);
        slangs.addAll(tmpSlangs);
        irregulars.addAll(tmpIrregulars);
        expressions.addAll(tmpExpressions);
    }


    public AnalysedText(String downloadedText){
        try {
            JSONObject obj = new JSONObject(downloadedText);

            JSONArray j_repeated = obj.getJSONArray("repeated");
            for(int i = 0; i < j_repeated.length(); i++){
                repeated.add(j_repeated.getString(i));
            }

            JSONArray j_archaisms = obj.getJSONArray("archaisms");
            for(int i = 0; i < j_archaisms.length(); i++){
                archaisms.add(j_archaisms.getString(i));
            }


            JSONArray j_slangs = obj.getJSONArray("slangs");
            for(int i = 0; i < j_slangs.length(); i++){
                slangs.add(j_slangs.getString(i));
            }


            JSONArray j_irregulars = obj.getJSONArray("irregulars");
            for(int i = 0; i < j_irregulars.length(); i++){
                irregulars.add(j_irregulars.getString(i));
            }


            JSONArray j_expressions = obj.getJSONArray("expressions");
            for(int i = 0; i < j_expressions.length(); i++){
                expressions.add(j_expressions.getString(i));
            }

        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public AnalysedText(HashSet<String> tmpRepeated)
    {
        repeated.addAll(tmpRepeated);
    }


    public HashSet<String> getRepeated() {
        return repeated;
    }

    public void setRepeated(HashSet<String> _repeated) {
        this.repeated = _repeated;
    }

    public HashSet<String> getArchaisms() {
        return archaisms;
    }

    public void setArchaisms(HashSet<String> _archaisms) {
        this.archaisms = _archaisms;
    }

    public HashSet<String> getSlangs() {
        return slangs;
    }

    public void setSlangs(HashSet<String> _slangs) {
        this.slangs = _slangs;
    }

    public HashSet<String> getIrregulars() {
        return irregulars;
    }

    public void setIrregulars(HashSet<String> _irregulars) {
        this.irregulars = _irregulars;
    }

    public HashSet<String> getExpressions() {
        return expressions;
    }

    public void setExpressions(HashSet<String> _expressions) {
        this.expressions = _expressions;
    }

    @Override
    public String toString() {
        return "AnalysedText{" +
                "repeated=" + repeated +
                ", archaisms=" + archaisms +
                ", slangs=" + slangs +
                ", irregulars=" + irregulars +
                ", expressions=" + expressions +
                '}';
    }
}
